Sources/Datadog/Logging/LogOutputs/LogOutput.swift

```diff
@@ -15,5 +15,5 @@ internal struct LogAttributes {
 
 /// Type writting logs to some destination.
 internal protocol LogOutput {
-    func writeLogWith(level: LogLevel, message: String, date: Date, attributes: LogAttributes, tags: Set<String>)
+    func writeLogWith(level: LogLevel, message: String, error: Error?, date: Date, attributes: LogAttributes, tags: Set<String>)
```

Comment:
If we change `error: Error?` to `dderror: DDError?` here, we won't need to play with the vague `Error` representation downstream (it will be mapped to `DDError` early in the `Logger`). This will also enable seamless integration of crash reporting (`RUMM-1050`), as the `DDError()` for crash report can be easily constructed.